hydra: cannot build with PBS support

I tried to configure MPICH with `./configure --with-device=ch4:ucx --with-ucx=/apps/ucx/1.12.0 --with-pbs=/opt/pbs/default`, but it fails with:

```
checking for tm.h... yes
checking for tm_init in -ltorque... no
configure: error: --with-pbs is given but not found
```

This cluster uses PBS Pro... is that supported, or does PBS support in MPICH actually mean PBS/torque?

It configures successfully:
```
checking tm.h usability... yes
checking tm.h presence... yes
checking for tm.h... yes
checking for library containing tm_init... -lpbs
```

but it also emits a spurious (?) warning message:
`configure: WARNING: unrecognized options: --with-pbs`

The build then fails:
```
gcc: error: pbs: No such file or directory
gcc: error: pbs: No such file or directory
gcc: error: pbs: No such file or directory
make[3]: *** [Makefile:1124: hydra_nameserver] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [Makefile:1137: hydra_persist] Error 1
make[3]: *** [Makefile:1157: hydra_pmi_proxy] Error 1
gcc: error: pbs: No such file or directory
make[3]: *** [Makefile:1185: mpiexec.hydra] Error 1
```

--- a/confdb/aclocal_libs.m4
+++ b/confdb/aclocal_libs.m4
@@ -113,13 +113,16 @@ AC_DEFUN([PAC_CHECK_HEADER_LIB_OPTIONAL],[
         pac_have_$1=no
     else
         dnl Other than "embedded" or "no", we check ...
-        PAC_CHECK_HEADER_LIB($2,$3,$4,pac_have_$1=yes,pac_have_$1=no,$5)
+        for a in $3 ; do
+            PAC_CHECK_HEADER_LIB($2,$a,$4,pac_have_$1=yes,pac_have_$1=no,$5)
+            if test "$pac_have_$1" = "yes"; then
+                PAC_LIBS_ADD(-l$a)
+                break
+            fi
+        done
         if test "${pac_have_$1}" = "no" -a -n "${with_$1}" ; then
             dnl user asks for it, so missing is an error
             AC_MSG_ERROR([--with-$1 is given but not found])
-        fi
-        if test "${pac_have_$1}" = "yes" ; then
-            PAC_LIBS_ADD(-l$3)
         fi
     fi
 ])

--- a/src/pm/hydra/configure.ac
+++ b/src/pm/hydra/configure.ac
@@ -253,7 +253,7 @@ for hydra_bss_name in ${hydra_bss_names}; do
 	pbs)
 		hydra_bss_external=true
 		# Check if TM library support is available
-		PAC_CHECK_HEADER_LIB_OPTIONAL(pbs, tm.h, torque, tm_init)
+		PAC_CHECK_HEADER_LIB_OPTIONAL(pbs, tm.h, [torque pbs], tm_init)
                 if test "$pac_have_pbs" = "yes" ; then
 		    available_launchers="$available_launchers pbs"
                 fi

